pkg/controller/certificatemanager/controller.go

```diff
@@ -0,0 +1,308 @@
+
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+
+	cr.SetConditions(runtimev1alpha1.Available())
+
+	tags, err := e.client.ListTagsRequest(&awsacmpca.ListTagsInput{
+		CertificateAuthorityArn: aws.String(cr.Status.AtProvider.CertificateAuthorityArn),
+	}).Send(ctx)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errListTagsFailed)
+	}
+
+	upToDate := acmpca.IsCertificateAuthorityUpToDate(cr, certificateAuthority, tags.Tags)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errUpToDateFailed)
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: upToDate,
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+
+	cr, ok := mgd.(*v1alpha1.CertificateAuthority)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	response, err := e.client.CreateCertificateAuthorityRequest(acmpca.GenerateCreateCertificateAuthorityInput(&cr.Spec.ForProvider)).Send(ctx)
+
+	if response != nil {
+
+		cr.Status.AtProvider.CertificateAuthorityArn = aws.StringValue(response.CreateCertificateAuthorityOutput.CertificateAuthorityArn)
+		cr.Status.AtProvider.RenewalPermission = cr.Spec.ForProvider.CertificateRenewalPermissionAllow
+
+		if cr.Spec.ForProvider.CertificateRenewalPermissionAllow {
+
+			_, err = e.client.CreatePermissionRequest(&awsacmpca.CreatePermissionInput{
+
+				Actions:                 []awsacmpca.ActionType{awsacmpca.ActionTypeIssueCertificate, awsacmpca.ActionTypeGetCertificate, awsacmpca.ActionTypeListPermissions},
+				CertificateAuthorityArn: aws.String(cr.Status.AtProvider.CertificateAuthorityArn),
+				Principal:               aws.String("acm.amazonaws.com"),
+			}).Send(ctx)
+
+		}
+	}
+
+	return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) { // nolint:gocyclo
+
+	cr, ok := mgd.(*v1alpha1.CertificateAuthority)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	response, err := e.client.DescribeCertificateAuthorityRequest(&awsacmpca.DescribeCertificateAuthorityInput{
+		CertificateAuthorityArn: aws.String(cr.Status.AtProvider.CertificateAuthorityArn),
+	}).Send(ctx)
+
+	if response != nil && response.CertificateAuthority.Status != awsacmpca.CertificateAuthorityStatusPendingCertificate {
+		_, err = e.client.UpdateCertificateAuthorityRequest(acmpca.GenerateUpdateCertificateAuthorityInput(cr)).Send(ctx)
+	}
+
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errCertificateAuthority)
+	}
+
+	if len(cr.Spec.ForProvider.Tags) > 0 {
```

Comment:
We generally do checks like this in the `client` package.

Reply:
As we are looking for any change we have to check with the cluster resource specification.